### PR TITLE
Add Edge Identity test and Update Readme files

### PR DESCRIPTION
**Pull request by cacheung**
Add Edge Identity test and Update Readme files

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: example/lib/edge.dart [hunk 1]
```diff
@@ -52,7 +52,6 @@ class _MyAppState extends State<EdgePage> {
   Future<void> sendEvent([datasetId]) async {
     late List<EventHandle> result;
     Map<String, dynamic> xdmData = {"eventType": "SampleEventType"};
-    Map<String, dynamic> data = {"free": "form", "data": "example"};
```
Comment (emdobrin) [inline example/lib/edge.dart:55]: should we rather keep this and send it under the "data" key below?

File: plugins/flutter_aepedge/README.md [hunk 2]
```diff
@@ -21,8 +21,9 @@ Install instructions for this package can be found [here](https://pub.dev/packag
 
 Run:
 
-```bash
-flutter test
+```
+$ cd plugins/flutter_aepedge/
```
Comment (emdobrin) [inline plugins/flutter_aepedge/README.md:25]: this readme seems to be under the plugins/flutter_aepedge space, so I think this line is unnecessary.
this comment applies for the other extensions too

File: plugins/flutter_aepedge/README.md [hunk 3]
```diff
@@ -102,10 +103,13 @@ public class MainApplication extends FlutterApplication {
 In your Flutter application, import the Edge extension as follows:
 ```dart
 import 'package:flutter_aepedge/flutter_aepedge.dart';
+import 'package:flutter_aepedge/flutter_aepedge_data.dart';
```
Comment (emdobrin) [inline plugins/flutter_aepedge/README.md:106]: Why is this required? I see flutter_aepedge.dart imports the data too - https://github.com/adobe/aepsdk_flutter/blob/staging/plugins/flutter_aepedge/lib/flutter_aepedge.dart

File: plugins/flutter_aepedgeconsent/README.md [hunk 4]
```diff
@@ -120,7 +123,8 @@ static Future<String> get extensionVersion
 String version = await Consent.extensionVersion;
 ```
 ------
-### Getting Current Consent Preference:
+### getConsents:
```
Comment (emdobrin) [inline plugins/flutter_aepedgeconsent/README.md:126]: let's remove the ':' after each API name in the headers

File: plugins/flutter_aepedgeconsent/README.md [hunk 5]
```diff
@@ -110,6 +111,8 @@ import 'package:flutter_aepedgeconsent/flutter_aepedgeconsent.dart';
 ------
 ## API reference
 ### extensionVersion
+Returns the SDK version of the Identity for Edge Consent extension
```
Comment (emdobrin) [inline plugins/flutter_aepedgeconsent/README.md:114]: Returns the SDK version of the Consent extension.

File: plugins/flutter_aepedge/pubspec.yaml [hunk 6]
```diff
@@ -10,6 +10,7 @@ environment:
   flutter: ">=2.0.0"
 
 dependencies:
+#TODO
```
Comment (emdobrin) [inline plugins/flutter_aepedge/pubspec.yaml:13]: edge depends on edgeidentity too

File: plugins/flutter_aepedgeidentity/README.md [hunk 7]
```diff
@@ -1,6 +1,356 @@
...
 
-`flutter_aepedgeidentity` is a flutter plugin for the iOS and Android [AEPEdgeIdentity SDK](https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network) to allow for integration with Flutter applications. Functionality to enable the Edge Identity extension is provided entirely through Dart documented below.
\ No newline at end of file
+`flutter_aepedgeidentity` is a flutter plugin for the iOS and Android [AEPEdgeIdentity SDK](https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network) to allow for integration with Flutter applications. Functionality to enable the Edge Identity extension is provided entirely through Dart documented below.
+
+## Prerequisites
+
+The Adobe Experience Platform Identity for Edge Network extension has the following peer dependency, which must be installed prior to installing the identity extension:
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:9]: prior to installing it:

File: plugins/flutter_aepedgeidentity/README.md [hunk 8]
```diff
@@ -1,6 +1,356 @@
...
+@implementation AppDelegate
+
+// AppDelegate.m
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    [AEPMobileCore setWrapperType:AEPWrapperTypeFlutter];
+
+     // TODO: Set up the preferred Environment File ID from your mobile property configured in Data Collection UI
+    NSString* ENVIRONMENT_FILE_ID = @'YOUR-APP-ID';
+    
+    const UIApplicationState appState = application.applicationState;
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:57]: not used in this example, let's remove

File: plugins/flutter_aepedgeidentity/README.md [hunk 9]
```diff
@@ -1,6 +1,356 @@
...
+### Importing the extension
+In your Flutter application, import the Edge Identity extension as follows:
+```dart
+import 'package:flutter_aepedgeidentity/flutter_aepedgeidentity.dart';
+import 'package:flutter_aepedgeidentity/flutter_aepedgeidentity_data.dart';
+```
+------
+## API reference
+### extensionVersion
+Returns the SDK version of the Identity for Edge Network extension
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:109]: Add a period at the end of each sentence.

File: plugins/flutter_aepedgeidentity/README.md [hunk 10]
```diff
@@ -1,6 +1,356 @@
...
+static Future<String> get extensionVersion
+ ```
+
+**Example**
+ ```dart
+String version = await Identity.extensionVersion;
+ ```
+------
+### getExperienceCloudId
+Returns the Experience Cloud ID. An empty string is returned if the Experience Cloud ID was previously cleared.
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:122]: for all apis can use the description from https://aep-sdks.gitbook.io/docs/foundation-extensions/identity-for-edge-network/api-reference#getexperiencecloudid instead

File: plugins/flutter_aepedgeidentity/README.md [hunk 11]
```diff
@@ -1,6 +1,356 @@
...
+String result = await Identity.getExperienceCloudId;
+```
+------
+### getIdentities
+Returns all identifiers, including customer identifiers which were previously added.
+If there are no identifiers stored in the `Identity` extension, then an empty `IdentityMap` is returned.
+
+**Syntax**
+```dart
+static Future<IdentityMap> get getIdentities
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:140]: We should update the APIs here and in the dart file and drop "get" from the API name, so it reads like:

static Future<String> get identities
static Future<String> get experienceCloudId
static Future<String> get urlVariables

File: plugins/flutter_aepedgeidentity/README.md [hunk 12]
```diff
@@ -1,6 +1,356 @@
...
+The possible authenticated states are:
+* AMBIGUOUS - the state is ambiguous or not defined
+* AUTHENTICATED - the user is identified by a login or similar action
+* LOGGED_OUT - the user was identified by a login action at a previous time, but is not logged in now
+
+**Syntax**
+
+```dart
+enum AuthenticatedState { AUTHENTICATED, LOGGED_OUT, AMBIGUOUS }
+}
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:341]: remove accolade and close sample code

**Review by emdobrin — COMMENTED** [review]
Docs review

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+      //test an invalid namespace
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
+
+      //test an empty namespace
+      final int actualItemsForNamespaceEmptyLength =
+          idMap.getIdentityItemsForNamespace('').length;
+      expect(actualItemsForNamespaceEmptyLength, equals(0));
+    });
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:88]: can we move each of these in their own test( ... ) under a new group   group('IdentityMap getIdentityItemsForNamespace', () {...})

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 14]
```diff
@@ -0,0 +1,228 @@
...
+          idMap.getIdentityItemsForNamespace('namespace1').length;
+
+      expect(actualSameIdSameItemLength, equals(1));
+      expect(actualSameIdSameItem[0].id, expectedSameIdSameItem[0].id);
+      expect(actualSameIdSameItem[0].authenticatedState,
+          expectedSameIdSameItem[0].authenticatedState);
+      expect(
+          actualSameIdSameItem[0].primary, expectedSameIdSameItem[0].primary);
+
+      //test add item with same id but different item, should replace existing item
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:134]: same here, I think you can move this under its own test(...), and you will need to reinitialize idMap in the setup step which runs before each test

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+      expect(actualItemsForNamespace2[0].authenticatedState,
+          expectedItemsForNamespace2[0].authenticatedState);
+      expect(actualItemsForNamespace2[0].primary,
+          expectedItemsForNamespace2[0].primary);
+
+      //test removeAllItems in namespace2, namespace2 should be removed
+      idMap.removeItem(item3, 'namespace2');
+      expect(
+          idMap.getIdentityItemsForNamespace('namespace2').length, equals(0));
+    });
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:201]: let's separate into its own test

**Comment by cacheung** [comment]
Ready to be reviewed again.

File: plugins/flutter_aepedge/README.md [hunk 16]
```diff
@@ -21,8 +21,8 @@ Install instructions for this package can be found [here](https://pub.dev/packag
 
 Run:
 
-```bash
-flutter test
+```
```
Comment (cdhoffmann) [inline plugins/flutter_aepedge/README.md:24]: Lets leave the bash syntax here.

File: plugins/flutter_aepedge/README.md [hunk 17]
```diff
@@ -182,6 +184,13 @@ final ExperienceEvent experienceEvent = ExperienceEvent({
   "datasetIdentifier": "datasetIdExample"
 });
 ```
+#### EventHandle
+The EventHandle is a response fragment from Adobe Experience Platform Edge Network for a sent XDM Experience Event. One event can receive none, one or multiple EdgeEventHandle(s) as response.
```
Comment (cdhoffmann) [inline plugins/flutter_aepedge/README.md:188]: ... as a response.

File: plugins/flutter_aepedgeconsent/README.md [hunk 18]
```diff
@@ -20,8 +20,8 @@ Install instructions for this package can be found [here](https://pub.dev/packag
 
 Run:
 
-```bash
-flutter test
+```
```
Comment (cdhoffmann) [inline plugins/flutter_aepedgeconsent/README.md:23]: keep bash syntax

File: plugins/flutter_aepedgeconsent/pubspec.yaml [hunk 19]
```diff
@@ -13,7 +13,6 @@ dependencies:
   flutter:
     sdk: flutter
   flutter_aepcore: ">= 1.0.0 <2.0.0"
-  #flutter_aepedgeidentity: ">= 1.0.0 <2.0.0"
```
Comment (cdhoffmann) [inline plugins/flutter_aepedgeconsent/pubspec.yaml:16]: Does this not depend on aepedgeidentity?

File: plugins/flutter_aepedgeidentity/README.md [hunk 20]
```diff
@@ -1,6 +1,362 @@
...
+Updating identities using a reserved namespace is not allowed using this API. The reserved namespaces are:
+
+* ECID
+* IDFA
+* GAID
+
+**Syntax**
+
+```dart
+static Future<IdentityMap> get getIdentities
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:243]: incorrect syntax here for updateidentities

File: plugins/flutter_aepedgeidentity/README.md [hunk 21]
```diff
@@ -1,6 +1,362 @@
...
+static Future<void> resetIdentities()
+```
+
+**Example**
+
+```dart
+MobileCore.resetIdentities()
+```
+------
+### setAdvertisingIdentifier:
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/README.md:212]: let's remove ":" for all apis

**Review by cdhoffmann — CHANGES_REQUESTED** [review]

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 22]
```diff
@@ -0,0 +1,262 @@
...
+    test('test an invalid namespace', () async {
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
+    });
+
+    test('test an empty namespace', () async {
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:119]: testing empty string here instead of invalid? 
actualItemsForNamespaceEmptyLength

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 23]
```diff
@@ -0,0 +1,262 @@
...
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return expectedItemsForNamespace1;
+      });
+    });
+
+    test('test a non-existed namespace', () async {
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:104]: non-existing

File: plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart [hunk 24]
```diff
@@ -0,0 +1,262 @@
...
+          expectedItemsForNamespace2[0].authenticatedState);
+      expect(actualItemsForNamespace2[0].primary,
+          expectedItemsForNamespace2[0].primary);
+    });
+    test('test removeAllItems in namespace2, namespace2 should be removed',
+        () async {
+      //test with getNamespaces()
+      idMap.removeItem(item3, 'namespace2');
+      expect(
+          idMap.getIdentityItemsForNamespace('namespace2').length, equals(0));
```
Comment (emdobrin) [inline plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart:234]: let's also include an assert for idMap.isEmpty true

File: plugins/flutter_aepedgeconsent/pubspec.yaml (hunk 19 above)
Comment (cacheung) [reply plugins/flutter_aepedgeconsent/pubspec.yaml:16]: Consent doesn't.  Only Edge Networks plugin depends on aepedgeidentity.

**Comment by cacheung** [comment]
@emdobrin @cdhoffmann, thanks for the review comments.  I updated them in the latest commit.

**Review by cdhoffmann — APPROVED** [review]
